test/helpers/utils.go

```diff
@@ -383,6 +386,36 @@ func MCLabelForWorkers() map[string]string {
 // 	}
 // }
 
+// Creates an Ign3 file whose contents are gzipped and encoded according to
+// https://datatracker.ietf.org/doc/html/rfc2397
+func CreateGzippedIgn3File(path, content string, mode int) (ign3types.File, error) {
+	ign3File := ign3types.File{}
+
+	buf := bytes.NewBuffer([]byte{})
+
+	gzipWriter := gzip.NewWriter(buf)
```

Comment:
Wonder if we should factor out a helper for here and https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update_test.go#L401

Reply:
In general, I'd like for us to refactor our test helpers and be a little more consistent with their usage.

Meanwhile, are you thinking about me refactoring the gzip portion of this into a helper?

Reply:
Discussed this privately with @mkenigs. We agreed that in the case mentioned above, it would make more sense for that test to migrate to use the `CreateGzippedIgn3File` helper introduced by this PR. However, this will be left to @mkenigs' discretion.